pkg/hostpath/hostpath.go

```diff
@@ -524,8 +528,27 @@ func mapBackingResourceToPod(hpv *hostPathVolume) error {
 	return nil
 }
 
+func (hp *hostPath) populateCache(hpv *hostPathVolume) error {
+	klog.V(4).Infof("populating object-cache before creating mount scaffold")
+	switch strings.TrimSpace(hpv.GetSharedDataKind()) {
+	case "ConfigMap":
+		return objcache.SetConfigMap(hp.kubeClient, hpv.GetSharedDataKey())
+	case "Secret":
```

Comment:
I am assuming something similar to what is done above for ConfigMaps is coming and will be here once WIP is removed

Reply:
Indeed. Please consider the latest changes, it encompasses Secrets as well.